Mathlib/CategoryTheory/LocallyCartesianClosed/Sections.lean

```diff
@@ -0,0 +1,172 @@
+   v               v
+  ⊤_ C    ---->  I ⟹ I
+```
+-/
+abbrev sectionsObj (X : Over I) : C :=
+  pullbackObj (ihom I |>.map X.hom) (curryId I)
+
+/-- The functoriality of `sectionsObj`. -/
+abbrev sectionsMap {X X' : Over I} (u : X ⟶ X') :
+    sectionsObj X ⟶ sectionsObj X' :=
+  pullbackMap _ _ _ _ (ihom I |>.map u.left) (𝟙 _) (𝟙 _)
+    (by simp [← Functor.map_comp]) (by cat_disch)
+
+@[simp]
+lemma sectionsMap_id {X : Over I} : sectionsMap (𝟙 X) = 𝟙 _ := by
+  cat_disch
+
+@[simp]
+lemma sectionsMap_comp {X X' X'' : Over I} (u : X ⟶ X') (v : X' ⟶ X'') :
+    sectionsMap (u ≫ v) = sectionsMap u ≫ sectionsMap v := by
+  cat_disch
+
+variable (I)
+
+/-- The functor mapping an object `X` in `C` to the object of sections of `X` over `I`. -/
+@[simps]
+def sections : Over I ⥤ C where
+  obj X := sectionsObj X
+  map u := sectionsMap u
+
+variable {I}
+
+open ChosenPullbacksAlong
+
+variable [BraidedCategory C]
+
+/-- The currying operation `Hom ((star I).obj A) X → Hom A (I ⟹ X.left)`. -/
+def sectionsCurry {X : Over I} {A : C} (u : (toOver I).obj A ⟶ X) :
+    A ⟶ (sections I).obj X :=
+  ChosenPullbacksAlong.lift (curry ((β_ I A).hom ≫ u.left)) (toUnit A) (by
+    rw [curry_natural_right, Category.assoc, ← Functor.map_comp, w, toOver_obj_hom,
+    ← curry_natural_right, toUnit_comp_curryId]
+    congr
+    simp [braiding_hom_snd])
+
+/-- The uncurrying operation `Hom A (section X) → Hom ((star I).obj A) X`. -/
+def sectionsUncurry {X : Over I} {A : C} (v : A ⟶ (sections I).obj X) :
+    (toOver I).obj A ⟶ X := by
+  let v₂ : A ⟶ (I ⟶[C] X.left) := v ≫ ChosenPullbacksAlong.fst (ihom I |>.map X.hom) (curryId I)
+  have comm : toUnit A ≫ (curryId I) = v₂ ≫ (ihom I).map X.hom := by
+    rw [IsTerminal.hom_ext isTerminalTensorUnit (toUnit A ) (v ≫ snd ..)]
+    simp [v₂, condition]
+  dsimp [curryId] at comm
+  have w' := (ihom.adjunction I).homEquiv_naturality_right_square _ _ _ _ comm
+  simp only [curriedTensor_obj_obj, curriedTensor_obj_map, curry,
+    Equiv.symm_apply_apply] at w'
+  exact Over.homMk ((β_ A I).hom ≫ uncurry v₂) (by
+    dsimp [uncurry] at *
+    rw [Category.assoc, ← w', whiskerLeft_toUnit_comp_rightUnitor_hom, braiding_hom_fst])
+
+@[simp]
+theorem sections_curry_uncurry {X : Over I} {A : C} {v : A ⟶ (sections I).obj X} :
+    sectionsCurry (sectionsUncurry v) = v := by
+  dsimp [sectionsCurry, sectionsUncurry]
+  let v₂ : A ⟶ (I ⟶[C] X.left) := v ≫ ChosenPullbacksAlong.fst (ihom I |>.map X.hom) (curryId I)
+  apply ChosenPullbacksAlong.hom_ext
+  · simp
+  · subsingleton
+
+@[simp]
+theorem sections_uncurry_curry {X : Over I} {A : C} {u : (toOver I).obj A ⟶ X} :
+    sectionsUncurry (sectionsCurry u) = u := by
+  dsimp [sectionsCurry, sectionsUncurry]
+  ext
+  simp
+
+open Adjunction
+
+/-- An auxiliary definition which is used to define the adjunction between the star functor
+and the sections functor. See starSectionsAdjunction`. -/
+def coreHomEquivToOverSections : CoreHomEquiv (toOver I) (sections I) where
```

Comment:
```suggestion
@[simps homEquiv]
def coreHomEquivToOverSections : CoreHomEquiv (toOver I) (sections I) where
```

This annotation is needed so that the simps lemmas for `toOverSectionsAdj` refer to `sectionCurry` and `sectionUncurry` instead of this def.